test/Writer.hs

```diff
@@ -6,43 +6,46 @@ module Writer
 ) where
 
 import Control.Arrow ((&&&))
-import qualified Control.Carrier.Writer.Strict as StrictWriterC
+import qualified Control.Carrier.Writer.Strict as WriterC
 import Control.Effect.Writer
 import qualified Control.Monad.Trans.RWS.Lazy as LazyRWST
 import qualified Control.Monad.Trans.RWS.Strict as StrictRWST
 import qualified Control.Monad.Trans.Writer.Lazy as LazyWriterT
 import qualified Control.Monad.Trans.Writer.Strict as StrictWriterT
 import Data.Bifunctor (first)
-import Data.Functor.Identity (Identity(..))
 import Data.Tuple (swap)
 import Gen
 import qualified Monad
+import qualified MonadFix
 import Test.Tasty
 import Test.Tasty.Hedgehog
 
 tests :: TestTree
 tests = testGroup "Writer"
-  [ test "WriterC (Strict)" StrictWriterC.runWriter
-  , test "WriterT (Lazy)"   (fmap swap . LazyWriterT.runWriterT)
-  , test "WriterT (Strict)" (fmap swap . StrictWriterT.runWriterT)
-  , test "RWST (Lazy)"      (runRWST LazyRWST.runRWST)
-  , test "RWST (Strict)"    (runRWST StrictRWST.runRWST)
+  [ testGroup "WriterC" $
+    [ testMonad
+    , testMonadFix
+    , testWriter
+    ] >>= ($ RunL WriterC.runWriter)
+  , testGroup "(,)"              $ testWriter (RunL pure)
```

Comment:
I completely missed this instance and it turns out it was broken! Hooray for tests.